pkg/gke/components/CruGKE.vue

```diff
@@ -0,0 +1,840 @@
+          }
+          this.nodePools.forEach((p) => {
+            if (!valid(p.name)) {
+              msg = this.t('validation.required', { key: this.t('gke.groupName.label') });
+            }
+          });
+
+          return msg;
+        }
+
+      };
+    },
+
+    // upstreamSpec will be null if the user created a cluster with some invalid options such that it ultimately fails to create anything in gke
+    // this allows them to go back and correct their mistakes without re-making the whole cluster
+    isNewOrUnprovisioned() {
+      return this.mode === _CREATE || !this.normanCluster?.gkeStatus?.upstreamSpec;
+    },
+
+    isEdit() {
+      return this.mode === _CREATE || this.mode === _EDIT;
+    },
+
+    doneRoute() {
+      return this.value?.listLocation?.name;
+    },
+
+    hasCredential() {
+      return !!this.config?.googleCredentialSecret;
+    },
+
+    clusterId(): String | null {
+      return this.value?.id || null;
+    },
+
+    canManageMembers(): Boolean {
+      return canViewClusterMembershipEditor(this.$store);
+    },
+
+    CREATE(): string {
+      return _CREATE;
+    },
+
+    VIEW(): string {
+      return _VIEW;
+    },
+
+    machineTypeOptions(): {label: string, kind?: string, value?: string, disabled?: boolean, [key: string]: any}[] {
+      const allTypes = this.machineTypesResponse?.items;
+      const out = [] as {label: string, kind?: string, value?: string, disabled?: boolean, [key: string]: any}[];
+
+      if (!allTypes) {
+        return out;
+      }
+
+      allTypes.forEach((type, i) => {
+        const group = (type.name || '').split('-').shift();
+        const lastGroup = (allTypes[i - 1]?.name || '').split('-').shift();
+
+        if (group && group !== lastGroup) {
+          out.push({
+            label:    group,
+            disabled: true,
+            kind:     'group'
+          });
+        }
+        out.push( {
+          value: type.name,
+          label: type.description ? `${ type.name } - ${ type.description }` : type.name,
+          ...type
+        });
+      });
+
+      return out;
+    }
+
+  },
+
+  watch: {
+    defaultImageType(neu) {
+      if (this.mode === _CREATE) {
+        this.nodePools.forEach((pool) => this.$set(pool, 'config.imageType', neu));
+      }
+    },
+    isAuthenticated() {
+      this.debouncedLoadGCPData();
+    },
+    'config.zone'() {
+      this.debouncedLoadGCPData();
+    },
+    'config.region'() {
+      this.debouncedLoadGCPData();
+    },
+  },
+
+  methods: {
+    loadGCPData() {
+      this.errors = [];
+      if (this.config.projectID && this.config.googleCredentialSecret) {
+        this.getMachineTypes();
+      }
+    },
+
+    getMachineTypes() {
+      this.loadingMachineTypes = true;
+      const zone = this.config.zone || this.config.locations?.[0];
+
+      getGKEMachineTypes(this.$store, this.config.googleCredentialSecret, this.config.projectID, { zone, region: this.config.region }).then((res) => {
```

Comment:
IMHO for this use case then/catch looks neater, but i think we're trying to avoid using this pattern. changing this to try/await/catch should still work (even where it's called doesn't `await getMachineTypes`)